File: src/Modelos/Proveedor.php
<?php

namespace SITCAV\Modelos;

use Illuminate\Database\Eloquent\Collection;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * @property-read int $id
 * @property string $rif
 * @property string $nombre
 * @property string $telefono
 * @property-read Estado $estado
 * @property-read ?Localidad $localidad
 * @property-read ?Sector $sector
 * @property-read Collection<Producto> $productos
 * @property-read Collection<Compra> $ventas
 */
final class Proveedor extends Model
{
  protected $table = 'proveedores';
  public $timestamps = false;

  /**
   * @return BelongsTo<Estado>
   */
  function estado(): BelongsTo
  {
    return $this->belongsTo(Estado::class, 'id_estado');
  }

  /**
   * @return BelongsTo<Localidad>
   */
  function localidad(): BelongsTo
  {
    return $this->belongsTo(Localidad::class, 'id_localidad');
  }

  /**
   * @return BelongsTo<Sector>
   */
  function sector(): BelongsTo
  {
    return $this->belongsTo(Sector::class, 'id_sector');
  }

  /**
   * @return HasMany<Producto>
   */
  function productos(): HasMany
  {
    return $this->hasMany(Producto::class, 'id_proveedor');
  }

  /**
   * @return HasMany<Compra>
   */
  function ventas(): HasMany
  {
    return $this->hasMany(Compra::class, 'id_proveedor');
  }
}

File: models/proveedor.py
from db import db


class Proveedor(db.Model):
    __tablename__ = "proveedores"

    id = db.Column(db.Integer, primary_key=True)
    id_estado = db.Column(db.Integer, db.ForeignKey("estados.id"), nullable=True)
    id_localidad = db.Column(db.Integer, db.ForeignKey("localidades.id"), nullable=True)
    id_sector = db.Column(db.Integer, db.ForeignKey("sectores.id"), nullable=True)
    nombre = db.Column(db.String(200), nullable=False)
    rif = db.Column(db.String(50), nullable=True)
    telefono = db.Column(db.String(20), nullable=True)
    direccion = db.Column(db.String(300), nullable=True)

    # Relaciones
    productos = db.relationship("Producto", backref="proveedor", lazy=True)
    compras = db.relationship("Compra", backref="proveedor", lazy=True)

    def to_dict(self):
        return {
            "id": self.id,
            "nombre": self.nombre,
            "rif": self.rif,
            "telefono": self.telefono,
            "direccion": self.direccion,
        }
